tracer/src/Datadog.Trace/ContinuousProfiler/ProfilerStatus.cs

```diff
@@ -36,6 +38,12 @@ public ProfilerStatus()
                     // instead of "true" to avoid starting the profiler immediately after the installation
                     _isProfilingEnabled = manualDeployement.ToBoolean() ?? (manualDeployement == "auto");
                 }
+                else
+                {
+                    // the profiler is declared "enabled" just if the SSI environment variable exists to be sure that telemetry metrics
+                    // will contain the right status (i.e. we need the tracer to send the spans even if the profiler is not started yet)
+                    _isProfilingEnabled = (EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.SsiDeployed) != null);
```

Comment:
This doesn't match with the check in `TracerSettings`, right? There we check for `profiler` being present...

It feels like we're duplicating the activation logic in a couple of places, inconsistently... Feels like we should try to extract that somewhere? 🤔